.github/workflows/ci-checks.yaml

```diff
@@ -30,27 +30,46 @@ jobs:
       max-parallel: 4
       matrix:
         python-version: [3.7]
-      fail-fast: false 
+        java-version: ["8"]
+      fail-fast: false
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
+      - uses: actions/setup-java@v2
+        with:
+          distribution: "adopt"
+          java-version: ${{ matrix.java-version }}
       - name: Run image
         uses: abatilo/actions-poetry@v2.0.0
         with:
-          poetry-version: 1.1.11
+          poetry-version: 1.3.2
       - name: Install dependencies
         run: poetry install
       - name: test
-        run: cd tests && poetry run python -m unittest
+        run: cd tests && poetry run pytest --tap-stream --cov-report=html --junitxml=junit.xml
+      - name: Upload Test report
+        uses: dorny/test-reporter@v1
+        if: success() || failure()
+        with:
+          name: test-results
+          path: tests/junit.xml
+          reporter: java-junit
```

Comment:
This is fine, we are simply claiming to be junit because we are junit compatible.